spring-core/src/test/java/org/togglz/spring/activation/SpringEnvironmentPropertyActivationStrategyTest.java

```diff
@@ -0,0 +1,119 @@
+ * </p>
+ *
+ * @author Alasdair Mercer
+ */
+@RunWith(Theories.class)
+public class SpringEnvironmentPropertyActivationStrategyTest {
+
+    @DataPoints
+    public static final boolean[] DATA_POINTS = { true, false };
+
+    @Mock
+    private ApplicationContext mockApplicationContext;
+    @Mock
+    private Environment mockEnvironment;
+
+    private SpringEnvironmentPropertyActivationStrategy strategy;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+
+        when(mockApplicationContext.getEnvironment()).thenReturn(mockEnvironment);
+
+        strategy = new SpringEnvironmentPropertyActivationStrategy();
+    }
+
+    @After
+    public void tearDown() {
+        ContextClassLoaderApplicationContextHolder.release();
+    }
+
+    @Test
+    public void testGetId() {
+        assertEquals(SpringEnvironmentPropertyActivationStrategy.ID, strategy.getId());
+    }
+
+    @Test
+    public void testGetName() {
+        assertEquals(SpringEnvironmentPropertyActivationStrategy.NAME, strategy.getName());
+    }
+
+    @Theory
+    public void testIsActiveWithNoParam(boolean enabled) {
+        FeatureState featureState = new FeatureState(TestFeatures.FEATURE_ONE, !enabled);
+
+        ContextClassLoaderApplicationContextHolder.bind(mockApplicationContext);
+
+        when(mockEnvironment.getProperty(
+            "org.togglz.spring.activation.SpringEnvironmentPropertyActivationStrategyTest$TestFeatures.FEATURE_ONE",
+            Boolean.TYPE, false)).thenReturn(enabled);
+
+        assertEquals(enabled, strategy.isActive(featureState, null));
+    }
+
+    @Theory
+    public void testIsActiveWithParam(boolean enabled) {
+        String paramValue = "foo";
+        FeatureState featureState = new FeatureState(TestFeatures.FEATURE_ONE, !enabled);
+        featureState.setParameter(SpringEnvironmentPropertyActivationStrategy.PARAM_NAME, paramValue);
+
+        ContextClassLoaderApplicationContextHolder.bind(mockApplicationContext);
+
+        when(mockEnvironment.getProperty(paramValue, Boolean.TYPE, false)).thenReturn(enabled);
+
+        assertEquals(enabled, strategy.isActive(featureState, null));
+    }
+
+    @Test
+    public void testIsActiveWithNoApplicationContext() {
+        FeatureState featureState = new FeatureState(TestFeatures.FEATURE_ONE, true);
+
+        assertFalse(strategy.isActive(featureState, null));
+    }
+
+    @Test
+    public void testGetParameters() {
+        Parameter[] parameters = strategy.getParameters();
+
+        assertEquals(1, parameters.length);
+
+        Parameter parameter = parameters[0];
+
+        assertNotNull(parameter);
+        assertEquals(SpringEnvironmentPropertyActivationStrategy.PARAM_NAME, parameter.getName());
+        assertTrue(parameter.isOptional());
+        assertTrue(Strings.isNotBlank(parameter.getLabel()));
+        assertTrue(Strings.isNotBlank(parameter.getDescription()));
+    }
+
+    public enum TestFeatures implements Feature {
+
```

Comment:
Maybe the code formatter I created isn't so optimal. I'm not using Eclipse any more these days, so I didn't notice. But this isn't a blocker.